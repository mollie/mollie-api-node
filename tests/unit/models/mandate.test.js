import Mandate from 'models/mandate';

describe('mandate model', () => {
  it('should instantiate with default values', () => {
    const mandate = new Mandate();

    expect(mandate.isValid()).toBe(false);
    expect(mandate.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const mandateProps = JSON.parse(`{
        "resource": "mandate",
        "id": "mdt_h3gAaD5zP",
        "status": "valid",
        "method": "directdebit",
        "details": {
            "consumerName": "John Doe",
            "consumerAccount": "NL55INGB0000000000",
            "consumerBic": "INGBNL2A"
        },
        "mandateReference": "YOUR-COMPANY-MD1380",
        "signatureDate": "2018-05-07",
        "createdAt": "2018-05-07T10:49:08+00:00",
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT/mandates/mdt_h3gAaD5zP",
                "type": "application/hal+json"
            },
            "customer": {
                "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT",
                "type": "application/hal+json"
            },
            "documentation": {
                "href": "https://docs.mollie.com/reference/v2/mandates-api/get-mandate",
                "type": "text/html"
            }
        }
    }`);
    const mandate = new Mandate(mandateProps);

    expect(mandate.isValid()).toBe(true);
    expect(mandate.toPlainObject()).toMatchSnapshot();
  });
});
