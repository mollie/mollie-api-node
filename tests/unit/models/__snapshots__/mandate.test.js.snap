// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mandate model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "customer": null,
  },
  "createdAt": null,
  "details": null,
  "id": null,
  "mandateReference": null,
  "method": null,
  "resource": "mandate",
  "signatureDate": null,
  "status": null,
}
`;

exports[`mandate model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "customer": Object {
      "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT",
      "type": "application/hal+json",
    },
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/mandates-api/get-mandate",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT/mandates/mdt_h3gAaD5zP",
      "type": "application/hal+json",
    },
  },
  "createdAt": "2018-05-07T10:49:08+00:00",
  "details": Object {
    "consumerAccount": "NL55INGB0000000000",
    "consumerBic": "INGBNL2A",
    "consumerName": "John Doe",
  },
  "id": "mdt_h3gAaD5zP",
  "mandateReference": "YOUR-COMPANY-MD1380",
  "method": "directdebit",
  "resource": "mandate",
  "signatureDate": "2018-05-07",
  "status": "valid",
}
`;
